### PR TITLE
Use native debug symbols for Windows builds instead of breakpad-converted.

### PR DESCRIPTION
**Pull request by goodov**
This PR changes the way we prepare symbols archive for Windows builds.
Now we generate a valid symsrv-like directory structure with exe, dll, pdb to send as archive into Backtrace and roll it later to our own symbol server.

This PR also enables source indexer that modifies PDBs so Visual Studio can automatically download source files from GitHub during a crash dump debugging session.

Resolves https://github.com/brave/brave-browser/issues/26890

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by bridiver — APPROVED** [review]

File: chromium_src/tools/symsrc/source_index.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                       follow_junctions):
+    # Replace 'python3' executalbe with 'py -3', because it's installed with the
+    # official Python3 installer. 'python3' is not used on Windows.
+    python3_line_idx = 8
+    python3_replacement = ('python3 -c', 'py -3 -c')
+    if python3_replacement[0] in output_lines[python3_line_idx]:
+        output_lines[python3_line_idx] = output_lines[python3_line_idx].replace(
+            *python3_replacement)
+    else:
+        assert python3_replacement[1] in output_lines[python3_line_idx]
```
Comment (atuchin-m) [inline chromium_src/tools/symsrc/source_index.py:38]: Could we add more context in this assert?

File: tools/win/generate_symbols.py [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+
+    def add_if_has_pdb(file, should_exist=False):
+        file = os.path.join(args.build_dir, file)
+        if not os.path.exists(file):
+            if should_exist:
+                raise RuntimeError(f'{file} not found')
+            return
+        pdb_file = file + '.pdb'
+        if not os.path.exists(pdb_file):
+            return
```
Comment (atuchin-m) [inline tools/win/generate_symbols.py:101]: Is it a usual case? Or better to make it assert/RuntimeError ?

File: tools/win/generate_symbols.py [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+    fingerprint, filename = output.strip().split(' ', 1)
+    return fingerprint, filename
+
+
+async def get_pdb_fingerprint(pdb_path):
+    assert os.path.isabs(pdb_path)
+    llvm_pdbutil_path = os.path.join(ROOT_DIR, 'third_party', 'llvm-build',
+                                     'Release+Asserts', 'bin',
+                                     'llvm-pdbutil.exe')
+
```
Comment (atuchin-m) [inline tools/win/generate_symbols.py:188]: nit: add `assert os.path.exists(llvm_pdbutil_path)`

**Review by atuchin-m — APPROVED** [review]

File: tools/win/generate_symbols.py (hunk 2 above)
Comment (goodov) [reply tools/win/generate_symbols.py:101]: It's an expected case. We process https://github.com/chromium/chromium/blob/116.0.5845.1/chrome/installer/mini_installer/chrome.release here, so we have `.pak`, `.json`, `.dat`, `.bin` files here.

Some `.dll`s may be prebuilt, so we can't assert `.pdb` existence either.

I'll add a sanity assert to make sure we have at least `chrome.dll.pdb`.

**Pull request merged**